pkg/client/client.go

```diff
@@ -180,77 +209,191 @@ func (o *OptimizelyClient) GetFeatureVariableBoolean(featureKey, variableKey str
+			}
+		}
+	}()
 	if err != nil {
 		return "", err
 	}
 	if valueType != entities.String {
 		return "", fmt.Errorf("variable value for key %s is wrong type", variableKey)
 	}
+
 	return value, err
 }
 
 // GetFeatureVariableJSON returns the feature variable value of type json associated with the given feature and variable keys.
 func (o *OptimizelyClient) GetFeatureVariableJSON(featureKey, variableKey string, userContext entities.UserContext) (value *optimizelyjson.OptimizelyJSON, err error) {
 
-	val, valueType, err := o.GetFeatureVariable(featureKey, variableKey, userContext)
+	val, valueType, featureDecision, err := o.getFeatureVariable(featureKey, variableKey, userContext)
+	defer func() {
+		if o.notificationCenter != nil {
+			var variableValue interface{}
+			if value != nil {
+				variableValue = value.ToMap()
+			} else {
+				variableValue = val
+			}
+			variableMap := map[string]interface{}{
+				"variableKey":   variableKey,
+				"variableType":  valueType,
+				"variableValue": variableValue,
+			}
+			decisionNotification := decision.FeatureNotificationWithVariables(featureKey, featureDecision, &userContext, variableMap)
+			decisionNotification.Type = notification.FeatureVariable
+
+			if e := o.notificationCenter.Send(notification.Decision, *decisionNotification); e != nil {
+				o.logger.Warning("Problem with sending notification")
+			}
+		}
+	}()
 	if err != nil {
 		return value, err
 	}
 
 	value, err = optimizelyjson.NewOptimizelyJSONfromString(val)
 	if err != nil || valueType != entities.JSON {
-		return nil, fmt.Errorf("variable value for key %s is invalid or wrong type", variableKey)
+		value, err = nil, fmt.Errorf("variable value for key %s is invalid or wrong type", variableKey)
 	}
 
 	return value, err
 }
 
 // GetFeatureVariable returns feature variable as a string along with it's associated type.
-func (o *OptimizelyClient) GetFeatureVariable(featureKey, variableKey string, userContext entities.UserContext) (value string, valueType entities.VariableType, err error) {
+func (o *OptimizelyClient) getFeatureVariable(featureKey, variableKey string, userContext entities.UserContext) (string, entities.VariableType, *decision.FeatureDecision, error) {
 
 	featureDecisionContext, featureDecision, err := o.getFeatureDecision(featureKey, variableKey, userContext)
 	if err != nil {
-		return "", "", err
+		return "", "", &featureDecision, err
 	}
 
 	variable := featureDecisionContext.Variable
 
 	if featureDecision.Variation != nil {
 		if v, ok := featureDecision.Variation.Variables[variable.ID]; ok && featureDecision.Variation.FeatureEnabled {
-			return v.Value, variable.Type, err
+			return v.Value, variable.Type, &featureDecision, nil
 		}
 	}
 
-	return variable.DefaultValue, variable.Type, err
+	return variable.DefaultValue, variable.Type, &featureDecision, nil
+}
+
+// GetFeatureVariable returns feature variable as a string along with it's associated type.
```

Comment:
This returns the converted value, not variable as a string.

Reply:
good catch, actually that was old definition and I have not changed it. I will fix it.